gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -1768,23 +1768,29 @@ TileDBDataset::Create( const char * pszFilename, int nXSize, int nYSize, int nBa
         for( int i = 0; i < poDS->nBands;i++ )
             poDS->SetBand( i+1, new TileDBRasterBand( poDS.get(), i+1 ) );
 
-        poDS->SetMetadataItem( "NBITS", 
-                CPLString().Printf( "%d", poDS->nBitsPerSample ),
-                "IMAGE_STRUCTURE" );
-        poDS->SetMetadataItem( "DATA_TYPE", 
-                CPLString().Printf( "%d", poDS->eDataType ),
-                "IMAGE_STRUCTURE" );
-
-        poDS->SetMetadataItem( "X_SIZE", CPLString().Printf( "%d", poDS->nRasterXSize ), "IMAGE_STRUCTURE" );
-        poDS->SetMetadataItem( "Y_SIZE", CPLString().Printf( "%d", poDS->nRasterYSize ), "IMAGE_STRUCTURE" );
+        char** papszImageStruct = nullptr;
+        papszImageStruct = CSLAddNameValue(papszImageStruct, "NBITS",
+                        CPLString().Printf( "%d", poDS->nBitsPerSample ) );
+        papszImageStruct = CSLAddNameValue(papszImageStruct, "DATA_TYPE",
+                        CPLString().Printf( "%d", poDS->eDataType ) );
```

Comment:
I didn't note that previously, but serializing the enumeration value as a integer is rather user (or perhaps here developer :-)) unfriendly. Maybe using GDALGetDataTypeName() would be more appropriate ?

Reply:
I have added backwards compatibility for 2.5.0 and changed the code to write strings for the data types.